src/reducers/weather_reducer.js

```diff
@@ -0,0 +1,35 @@
+import { FETCH_WEATHER, SET_ERROR } from "../actions";
+
+const initialState = {
+  weatherInfo: [],
+};
+
+const weatherReducer = function (state = initialState, action) {
+  switch (action.type) {
+    case FETCH_WEATHER: {
+      return {
+        ...state,
+        weatherInfo: [
+          { ...action.payload.data },
+          ...state.weatherInfo.filter(
+            (entry) => entry.city.id !== action.payload.data.city.id
+          ),
+        ],
```

Comment:
I'd recommend having `initialState` as an array and updating it with the payload data. As things stand, the `weatherInfo` key is a little redundant.